swiftpkg/internal/modulemap_parser/collect_module.bzl

```diff
@@ -91,15 +91,20 @@ def _process_module_tokens(parsed_tokens, prefix_tokens, is_submodule):
         return None, err
     consumed_count += 1
 
-    module_id_or_asterisk_token, err = tokens.get(parsed_tokens, 1, count = tlen)
+    module_declaration_identifier_token, err = tokens.get(parsed_tokens, 1, count = tlen)
     if err != None:
         return None, err
     consumed_count += 1
-    if not tokens.is_a(module_id_or_asterisk_token, tts.identifier) and \
-       not tokens.is_a(module_id_or_asterisk_token, tts.operator, operators.asterisk):
+
+    # Check if the module identifier is an identifier, asterisk, or string literal
+    # Examples: module Foo, module *, module "Foo/Bar.h"
```

Comment:
If I am reading [the spec](https://clang.llvm.org/docs/Modules.html#id27) properly, a `module-id` must be an `identifier`. I don't see that it allows a literal string. Can you explain why you think string literals should be supported here?

Reply:
It turns out that the spec is wrong 🫨 

https://github.com/llvm/llvm-project/blob/dafb90dedcda1ad7b94b0bcdbbe7478f7d0f31f6/clang/lib/Lex/ModuleMap.cpp#L1833

The ModuleMap lexical analysis accepts literal strings